lib/enumerations.rb

```diff
@@ -82,7 +82,7 @@ def add_enumeration(reflection)
     #
     def define_getter_method(reflection)
       define_method(reflection.name) do
-        reflection.enumerator_class.find(self[reflection.foreign_key])
+        reflection.enumerator_class.find(self[reflection.foreign_key]) || self[reflection.foreign_key]
```

Comment:
When the value is valid, the enum will be returned like before. If the value is invalid, the getter will return that invalid value. This can be useful if you want to inspect errors:
```ruby
class User < ActiveRecord::Base
  validates :role, inclusion: { in: Role.all }
  enumeration :role
end

> user = User.new(role: 'invalid_value')
> user.validate
> user.errors.details[:role]
=> [{:error=>:inclusion, :value=>"invalid_value"}]
> user.role
=> "invalid_value"
```

Otherwise, the attribute would be set to nil:
```ruby
> user = User.new(role: 'invalid_value')
> user.validate
> user.errors.details[:role]
=> [{:error=>:inclusion, :value=>nil}]
> user.role
=> nil
```

I chose the first approach because other Active Record validations also don't unset a value if it's invalid, e.g.:
```ruby
class User < ActiveRecord::Base
  validates :age, inclusion: { in: 21..50 }
end

> user = User.new(age: 15)
> user.validate
> user.errors.details[:age]
=> [{:error=>:inclusion, :value=>15}]
> user.age
=> 15
```

Reply:
👍🏼